src/store/modules/auth/actions.js

```
@@ -1,10 +1,115 @@
 import debug from 'debug'
 
 import router from '../../../router'
+import User from '../../../util/api/user'
 
 const logger = debug('app:store:auth:actions')
 
 export default {
+  // Called from App.vue, ONLY IF there's an authToken in the query string or in local storage.
+  //  This be either simple or savvy users--we find out which once we retrieve the user object
+  //  from the API. If it's a simple user, we register Web3 using Infura. If it's a savvy user
+  //  we attempt to register Web3 using their wallet provider. If there is no wallet provider,
+  //  we kick them back to the login page with an error.
+  LOGIN_FROM_CACHED_TOKEN({ commit, dispatch, state }, authToken) {
+    logger('LOGIN_FROM_CACHED_TOKEN action being executed')
+
+    commit('SET_AUTH_STATE', {
+      authToken,
+    })
+
+    return User.getUser()
+      .then((user) => {
+        commit('SET_USER', {
+          user,
+        })
+
+        if (user.type === 'simple') {
+          return dispatch('web3/REGISTER_INFURA_PROVIDER', null, { root: true })
+        }
+
+        return dispatch('web3/REGISTER_WALLET_PROVIDER', null, { root: true })
+          .then(() => {
+            commit('web3/SET_IS_POLLING', {
+              isPolling: true,
+            }, {
+              root: true,
+            })
+
+            // No need to block on the promise resolution of polling
+            dispatch('web3/POLL_WEB3', null, { root: true })
+          })
+      })
+      .then(() => {
+        return dispatch('UPDATE_CONTRACT_STATE')
+      })
+      .catch((error) => {
+        return dispatch('HANDLE_LOGIN_ERROR', error)
```

Typically you don't want to return a promise (or any value really) from a `.catch()` (because of how the resulting value can get passed into a subsequent `.then()`). In this instance I don't think it will cause a problem, but I'd probably go ahead and remove it since it gives you no benefits either.

---

After seeing `HANDLE_LOGIN_ERROR` below, I'm pretty sure you can remove the returns here. If it were me, I'd also move the `throw` inside `HANDLE_LOGIN_ERROR` up into this catch as well, but the way you have it should work as well.

```
.catch((error) => {
  dispatch('HANDLE_LOGIN_ERROR', error)
  throw error
})
```

(I just like it to be clear inside a catch that you want to bubble the error up the chain.)